add initial text

acutmore opened the pull request:
brain dump ( duplicate PR to the bloomberg one)

acutmore committed:
add initial text

nicolo-ribaudo reviewed (commented):
I only read the readme so far, but it looks good overall!

acutmore committed:
add tests

acutmore committed:
README clarifications

acutmore committed:
apply suggestion from code review

acutmore committed:
expand Q&A

acutmore committed:
update with suggestions

- CompositeKey construct no longer varargs
- `keyBy` Map/Set config only accepts function
- Add Map.usingKeys helper instead
- Make `Symbol.keyBy` a method rather than a getter

acutmore committed:
CompositeKey back to varargs

acutmore committed:
add Tuple to polyfill

acutmore committed:
polyfill: handle symbol keys

acutmore committed:
remove CompositeKey.isKey - hide branch check in toStringTag

acutmore committed:
update README

acutmore committed:
add Set polyfill

mhofman reviewed (commented):
An advantage of introducing all elements at once is that we can have stronger requirements on the values allowed for each feature, in particular require that `[Symbol.keyBy]()` returns a primitive or `CompositeKey`, that R&T similarly require every value to be a primitive or object with `Symbol.keyBy`, and possibly even that the `keyBy` option of Collections must return a primitive or `CompositeKey`.

As I mentioned last time, the program can still express all keying equivalence they wish were these restriction not present by simply wrapping a regular object they would have used directly, into a `CompositeKey` containing only that object.

acutmore committed:
README updates

acutmore committed:
Apply suggestions from code review

Co-authored-by: Jordan Harband <ljharb@gmail.com>
